dvc/scm/git/backend/gitpython.py

```diff
@@ -370,14 +370,15 @@ def set_ref(
         except GitCommandError as exc:
             raise SCMError(f"Failed to set ref '{name}'") from exc
 
-    def get_ref(
-        self, name: str, follow: Optional[bool] = True
-    ) -> Optional[str]:
+    def get_ref(self, name: str, follow: bool = True) -> Optional[str]:
         from git.exc import GitCommandError
 
         if name == "HEAD":
             try:
-                return self.repo.head.commit.hexsha
+                if follow or self.repo.head.is_detached:
+                    return self.repo.head.commit.hexsha
+                else:
+                    return f"refs/heads/{self.repo.active_branch}"
```

Comment:
A bit offtopic and not important in such a small `else` statement, but what do you guys think about enabling no-else-after-return check in our linters? I seem to stumble upon it regularly.
```suggestion
                return f"refs/heads/{self.repo.active_branch}"
```